server/integration-tests/auth_stress_test.hs

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env stack
+-- stack --stack-yaml ../../client-haskell/stack.yaml runghc --package QuickCheck --package quickcheck-text -- -Wall
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+{- | This test first sends a number of write requests that should be accepted by
+   the server, followed by the same number of requests that should be accepted.
+-}
+import Prelude hiding (fail)
+
+import Data.Foldable (traverse_)
+import Data.Text (Text)
+import Data.Text.Arbitrary ()
+import Icepeak.Client (Client (..), setAtLeaf)
+import System.Environment (getArgs)
+import System.Exit (exitFailure)
+import System.IO (hPutStrLn, stderr)
+import Test.QuickCheck.Arbitrary (arbitrary)
+import Text.Read (readEither)
+
+import qualified Data.ByteString as SBS
+import qualified Network.HTTP.Client as HTTP
+import qualified Test.QuickCheck.Gen as Gen
+
+main :: IO ()
+main = do
+  args <- getArgs
+  case args of
+    [count] -> either fail main' (readEither count)
+    _ -> fail "usage: stress_test.hs <count>"
+  where fail msg = do { hPutStrLn stderr msg; exitFailure }
+
+testToken :: SBS.ByteString
```

Comment:
I personally am I a fan of placing code after the place where it is called, so to get a top-down view of the code, i.e, on top is the overview of what's going on and the details follow down below. Thus `testToken` should go under `main'`.